controllers/front/createIntent.php

```diff
@@ -41,18 +41,35 @@ public function initContent()
             'createIntent - intiContent'
         );
 
-        $amount = Tools::ps_round(Tools::getValue('amount'));
-        $currency = Tools::getValue('currency');
-        $paymentOption = Tools::getValue('payment_option');
-        $paymentMethodId = Tools::getValue('id_payment_method');
+        try {
+            $cart = $this->context->cart;
+
+            $currency = new Currency($cart->id_currency);
+            $amount = $cart->getOrderTotal();
+            $amount = Tools::ps_round($amount, 2);
+            $amount = $this->module->isZeroDecimalCurrency($currency->iso_code) ? $amount : $amount * 100;
 
-        $intentData = $this->constructIntentData($amount, $currency, $paymentOption, $paymentMethodId);
+            $paymentOption = Tools::getValue('payment_option');
+            $paymentMethodId = Tools::getValue('id_payment_method');
 
-        $cardData = $this->constructCardData($paymentMethodId);
+            $intentData = $this->constructIntentData($amount, $currency, $paymentOption, $paymentMethodId);
 
-        $intent = $this->createIdempotencyKey($intentData);
+            $cardData = $this->constructCardData($paymentMethodId);
 
-        $this->registerStripeEvent($intent);
+            $intent = $this->createIdempotencyKey($intentData);
+
+            $this->registerStripeEvent($intent);
+        } catch (Exception $e) {
+            ProcessLoggerHandler::logError(
+                "Retrieve Stripe Account Error => ".$e->getMessage(),
+                null,
+                null,
+                'createIntent'
+            );
+            ProcessLoggerHandler::closeLogger();
+            http_response_code(400);
+            die('An unexpected problem has occurred. Please contact the support : https://addons.prestashop.com/en/contact-us?id_product=24922');
```

Comment:
$this->ajaxDie(); instead of die();
it could be intresting to add in the error : $e->getMessage()

Reply:
I removed $e->getMessage() in die() after Youssef asked because he says that the customer must not see this kind of message in front. But i log this error with process logger.